theora_image_transport/src/theora_publisher.cpp

```diff
@@ -69,81 +70,73 @@ TheoraPublisher::~TheoraPublisher()
   // Latching doesn't make a lot of sense with this transport. Could try to save the last keyframe,
   // but do you then send all following delta frames too?
-  latch = false;
-  typedef image_transport::SimplePublisherPlugin<theora_image_transport::Packet> Base;
-  Base::advertiseImpl(nh, base_topic, queue_size, user_connect_cb, user_disconnect_cb, tracked_object, latch);
-
-  // Set up reconfigure server for this topic
-  reconfigure_server_ = boost::make_shared<ReconfigureServer>(this->nh());
-  ReconfigureServer::CallbackType f = boost::bind(&TheoraPublisher::configCb, this, _1, _2);
-  reconfigure_server_->setCallback(f);
-}
-
-void TheoraPublisher::configCb(Config& config, uint32_t level)
-{
-  // target_bitrate must be 0 if we're using quality.
-  long bitrate = 0;
-  if (config.optimize_for == theora_image_transport::TheoraPublisher_Bitrate)
-    bitrate = config.target_bitrate;
-  bool update_bitrate = bitrate && encoder_setup_.target_bitrate != bitrate;
-  bool update_quality = !bitrate && ((encoder_setup_.quality != config.quality) || encoder_setup_.target_bitrate > 0);
-  encoder_setup_.quality = config.quality;
-  encoder_setup_.target_bitrate = bitrate;
-  keyframe_frequency_ = config.keyframe_frequency;
-  
-  if (encoding_context_) {
-    int err = 0;
-    // libtheora 1.1 lets us change quality or bitrate on the fly, 1.0 does not.
-#ifdef TH_ENCCTL_SET_BITRATE
-    if (update_bitrate) {
-      err = th_encode_ctl(encoding_context_.get(), TH_ENCCTL_SET_BITRATE, &bitrate, sizeof(long));
-      if (err)
-        ROS_ERROR("Failed to update bitrate dynamically");
-    }
-#else
-    err |= update_bitrate;
-#endif
+  // latch = false;
 
-#ifdef TH_ENCCTL_SET_QUALITY
-    if (update_quality) {
-      err = th_encode_ctl(encoding_context_.get(), TH_ENCCTL_SET_QUALITY, &config.quality, sizeof(int));
-      // In 1.1 above call will fail if a bitrate has previously been set. That restriction may
-      // be relaxed in a future version. Complain on other failures.
-      if (err && err != TH_EINVAL)
-        ROS_ERROR("Failed to update quality dynamically");
-    }
-#else
-    err |= update_quality;
-#endif
-
-    // If unable to change parameters dynamically, just create a new encoding context.
-    if (err) {
-      encoding_context_.reset();
-    }
-    // Otherwise, do the easy updates and keep going!
-    else {
-      updateKeyframeFrequency();
-      config.keyframe_frequency = keyframe_frequency_; // In case desired value was unattainable
-    }
-  }
+  typedef image_transport::SimplePublisherPlugin<theora_image_transport::msg::Packet> Base;
+  Base::advertiseImpl(node, base_topic, custom_qos);
 }
 
-void TheoraPublisher::connectCallback(const ros::SingleSubscriberPublisher& pub)
-{
-  // Send the header packets to new subscribers
-  for (unsigned int i = 0; i < stream_header_.size(); i++) {
-    pub.publish(stream_header_[i]);
-  }
-}
+  // TODO: this method should be called when configuration change through
```

Comment:
It would be good to add to this that this was dynamic reconfigure and should change to be parameters and subscribe to parameter change events to trigger the same logic.